app/internal/secrets/secrets.go

```diff
@@ -20,34 +21,46 @@ const (
 	cookieSessionKeys = "cookie-session-keys"
 )
 
-type secretsCache interface {
-	GetSecretString(secretId string) (string, error)
+type secretsManager interface {
+	GetSecretValue(ctx context.Context, params *secretsmanager.GetSecretValueInput, optFns ...func(*secretsmanager.Options)) (*secretsmanager.GetSecretValueOutput, error)
 }
 
 type Client struct {
-	cache secretsCache
+	svc secretsManager
+
+	mu    sync.Mutex
+	cache map[string]string
```

Comment:
The package we were using hasn't got support for the V2 SDK, and it looks a bit abandonware. I'm putting a ticket in to add a simple LRU cache to replace this (which was basically what that package did)